UI/Panels/Settings/PeripheralsPanel.cs

```diff
@@ -16,50 +18,81 @@ public PeripheralsPanel()
+
+            // Joysticks
+            JoystickMgr.GetJoysticks().ForEach(js => { listBoxJoysticks.Items.Add(js.Name, true); });
+            JoystickMgr.GetExcludedJoysticks().ForEach(js => { listBoxJoysticks.Items.Add(js.Name, false); });
+
+            // MidiBoards
+            MidiMgr.GetMidiBoards().ForEach(mb => { listBoxMidiBoards.Items.Add(mb.Name, true); });
+            MidiMgr.GetExcludedMidiBoards().ForEach(mb => { listBoxMidiBoards.Items.Add(mb.Name, false); });
         }
 
-        public void SaveSettings()
+        private bool UpdateExclusionList(List<string> exclusionList, CheckedListBox listBox)
         {
-            // Joystick activation changed
-            if (Properties.Settings.Default.EnableJoystickSupport != checkBoxJoystickSupport.Checked)
+            bool listChanged = false;
+            foreach (string item in listBox.Items)
             {
-                if (checkBoxJoystickSupport.Checked) 
+                if (!listBox.CheckedItems.Contains(item) && !exclusionList.Contains(item))
                 {
-                    JoystickMgr.Connect();
+                    // Add to exclusion List
+                    exclusionList.Add(item);
+                    listChanged = true;
                 }
-                else 
+                else if (listBox.CheckedItems.Contains(item) && exclusionList.Contains(item))
                 {
-                    JoystickMgr.Shutdown();
+                    // Remove from exclusion List
+                    exclusionList.Remove(item);
+                    listChanged = true;
                 }
-                Properties.Settings.Default.EnableJoystickSupport = checkBoxJoystickSupport.Checked;
+            }
+            return listChanged;
+        }
+
+        public void SaveSettings()
+        {
+            List<string> excludedBoards = Properties.Settings.Default.ExcludedMidiBoards.Split('§').ToList();
+            List<string> excludedJoysticks = Properties.Settings.Default.ExcludedJoysticks.Split('§').ToList();
+
+            bool excludedJoysticksChanged = UpdateExclusionList(excludedJoysticks, listBoxJoysticks);
+            bool excludedBoardsChanged = UpdateExclusionList(excludedBoards, listBoxMidiBoards);
+
+            bool enableJoysticksChanged = Properties.Settings.Default.EnableJoystickSupport != checkBoxJoystickSupport.Checked;
+            bool enableMidiChanged = Properties.Settings.Default.EnableMidiSupport != checkBoxMidiSupport.Checked;
+
+            Properties.Settings.Default.ExcludedMidiBoards = string.Join("§", excludedBoards);
+            Properties.Settings.Default.ExcludedJoysticks = string.Join("§", excludedJoysticks);
+            Properties.Settings.Default.EnableJoystickSupport = checkBoxJoystickSupport.Checked;
+            Properties.Settings.Default.EnableMidiSupport = checkBoxMidiSupport.Checked;
+
+            // Update joystick management
+            if (!checkBoxJoystickSupport.Checked && enableJoysticksChanged)
```

Comment:
Ideally these actions should not happen withing the settingsdialog
We have a settings changed event handler that could handle this. We can do that at a later point... but it makes sense as refactoring.

Reply:
Yes, with settings changed event is better. I have put it on my todo list for later.